File: majorProject/src/main/java/com/aocc/framework/implementation/AndroidInput.java
package com.aocc.framework.implementation;

import java.util.List;

import android.content.Context;
import android.os.Build.VERSION;
import android.view.View;

import com.aocc.framework.Input;
import com.aocc.framework.Viewport;

public class AndroidInput implements Input {
    private final TouchHandler touchHandler;

    public AndroidInput(Context context, View view, Viewport viewport) {
        RotationHandler accelHandler = new RotationHandler(context);

        if (VERSION.SDK_INT < 5) {
            touchHandler = new SingleTouchHandler(view, viewport);
        } else {
            touchHandler = new MultiTouchHandler(view, viewport);
        }
    }

    public void updateViewport(Viewport viewport) {
        touchHandler.setViewport(viewport);
    }

    @Override
    public boolean isTouchDown(int pointer) {
        return touchHandler.isTouchDown(pointer);
    }

    @Override
    public int getTouchX(int pointer) {
        return touchHandler.getTouchX(pointer);
    }

    @Override
    public int getTouchY(int pointer) {
        return touchHandler.getTouchY(pointer);
    }

    @Override
    public List<TouchEvent> getTouchEvents() {
        return touchHandler.getTouchEvents();
    }

    @Override
    public float getRotationX() {
        return RotationHandler.getRotationX();
    }

    @Override
    public float getRotationY() {
        return RotationHandler.getRotationY();
    }

    @Override
    public float getRotationZ() {
        return RotationHandler.getRotationZ();
    }
}

File: majorProject/src/main/kotlin/com/aocc/framework/implementation/AndroidInput.kt
package com.aocc.framework.implementation

import android.content.Context
import android.os.Build
import android.view.View
import com.aocc.framework.Input
import com.aocc.framework.Viewport

class AndroidInput(context: Context, view: View, viewport: Viewport) : Input {
    private val touchHandler: TouchHandler

    init {
        RotationHandler(context)

        touchHandler = if (Build.VERSION.SDK_INT < 5) {
            SingleTouchHandler(view, viewport)
        } else {
            MultiTouchHandler(view, viewport)
        }
    }

    fun updateViewport(viewport: Viewport) {
        touchHandler.setViewport(viewport)
    }

    override fun isTouchDown(pointer: Int): Boolean {
        return touchHandler.isTouchDown(pointer)
    }

    override fun getTouchX(pointer: Int): Int {
        return touchHandler.getTouchX(pointer)
    }

    override fun getTouchY(pointer: Int): Int {
        return touchHandler.getTouchY(pointer)
    }

    override val touchEvents: List<Input.TouchEvent>
        get() = touchHandler.touchEvents

    override fun getRotationX(): Float {
        return RotationHandler.getRotationX()
    }

    override fun getRotationY(): Float {
        return RotationHandler.getRotationY()
    }

    override fun getRotationZ(): Float {
        return RotationHandler.getRotationZ()
    }
}
